source/windows/secure_channel_tls_handler.c

```diff
@@ -1465,12 +1486,31 @@ static int s_handler_shutdown(
                 if (!outgoing_message || outgoing_message->message_data.capacity < output_buffer.cbBuffer) {
                     return aws_channel_slot_on_handler_shutdown_complete(slot, dir, aws_last_error(), true);
                 }
-                memcpy(outgoing_message->message_data.buffer, output_buffer.pvBuffer, output_buffer.cbBuffer);
 
-                /* we don't really care if this succeeds or not, it's just sending the TLS alert. */
-                if (aws_channel_slot_send_message(slot, outgoing_message, AWS_CHANNEL_DIR_WRITE)) {
+                struct alert_data *alert_data = aws_mem_calloc(slot->alloc, 1, sizeof(struct alert_data));
+                if (!alert_data) {
                     aws_mem_release(outgoing_message->allocator, outgoing_message);
+                    return aws_channel_slot_on_handler_shutdown_complete(slot, dir, aws_last_error(), true);
                 }
+
+                alert_data->allocator = slot->alloc;
+                alert_data->slot = slot;
+                alert_data->shutdown_error = error_code;
+                alert_data->shutdown_immediately = abort_immediately;
+
+                outgoing_message->on_completion = s_on_alert_sent;
+                outgoing_message->user_data = alert_data;
+
+                memcpy(outgoing_message->message_data.buffer, output_buffer.pvBuffer, output_buffer.cbBuffer);                
+                
+                /* we don't really care if this succeeds or not, it's just sending the TLS alert. */

```

Comment:
Comment says we don't care, but we are definitely changing the return value, so, maybe we do?